Docs/source/developers/testing.rst

```diff
@@ -47,97 +30,149 @@ You can run those locally before you push to save some time, install them once l
+
+  .. code-block:: sh
+
+        ctest --test-dir build -j 2
 
-Run the test suite locally
+* Run tests available for the current build options and output anything outputted by the test program if the test should fail:
+
+  .. code-block:: sh
+
+        ctest --test-dir build --output-on-failure
+
+* Run tests available for the current build options with verbose output:
+
+  .. code-block:: sh
+
+        ctest --test-dir build --verbose
+
+* Run tests matching the regular expression ``laser_acceleration``:
+
+  .. code-block:: sh
+
+        ctest --test-dir build -R laser_acceleration
+
+* Run tests except those matching the regular expression ``laser_acceleration``:
+
+  .. code-block:: sh
+
+        ctest --test-dir build -E laser_acceleration
+
+Once the execution of CTest is completed, you can find all files associated with each test in its corresponding directory under ``build/bin/``.
+For example, if you run the single test ``test_3d_laser_acceleration``, you can find all files associated with this test in the directory ``build/bin/test_3d_laser_acceleration/``.
+
+If you modify the code base locally and want to assess the effects of your code changes on the automated tests, you need to first rebuild WarpX including your code changes and then rerun CTest.
+
+How to add automated tests
 --------------------------
 
-Once your new feature is ready, there are ways to check that you did not break anything.
-WarpX has automated tests running every time a commit is added to an open pull request.
-The list of automated tests is defined in `./Regression/WarpX-tests.ini <https://github.com/ECP-WarpX/WarpX/blob/development/Regression/WarpX-tests.ini>`__.
+As mentioned above, the input files and scripts used by the automated tests can be found in the `Examples <https://github.com/ECP-WarpX/WarpX/tree/development/Examples>`__ directory, either under `Physics_applications <https://github.com/ECP-WarpX/WarpX/tree/development/Examples/Physics_applications>`__ or `Tests <https://github.com/ECP-WarpX/WarpX/tree/development/Examples/Tests>`__.
 
-For easier debugging, it can be convenient to run the tests on your local machine by executing the script
-`./run_test.sh <https://github.com/ECP-WarpX/WarpX/blob/development/run_test.sh>`__ from WarpX's root folder, as illustrated in the examples below:
+Each test directory must contain a file named ``CMakeLists.txt`` where all tests associated with the input files and scripts in that directory must be listed.
 
-.. code-block:: sh
+A new test can be added by adding a corresponding entry in ``CMakeLists.txt`` as illustrated in the examples below:
 
-    # Example:
-    # run all tests defined in ./Regression/WarpX-tests.ini
-    ./run_test.sh
+* Add the **regular test** ``test_1d_laser_acceleration``, with input file and analysis script in ``Examples/Tests/laser_acceleration/``:
 
-    # Example:
-    # run only the test named 'pml_x_yee'
-    ./run_test.sh pml_x_yee
+  .. code-block:: sh
```

Comment:
CMake is a supported syntax :)
```suggestion
  .. code-block:: cmake
```

Reply:
Done in [8e64870](https://github.com/ECP-WarpX/WarpX/pull/5068/commits/8e64870fe4fe42fd6c88f69741fe7761d30226c6).